### PR TITLE
Implement log file download

### PR DESCRIPTION
**Pull request by cb341**
TICKET-24979

CleanShot 2026-04-30 at 15 51 11

### CODE REVIEW

File: app/helpers/onlylogs/application_helper.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 module Onlylogs
   module ApplicationHelper
+    def log_file_label(file)
```
Comment (cb341) [inline app/helpers/onlylogs/application_helper.rb:3]: I don't see the need to differentiate between relative and absolute path here..

File: app/views/onlylogs/logs/index.html.erb [hunk 2]
```diff
@@ -14,21 +14,26 @@
     .log-file-dropdown {
         margin-left: 0.5rem;
     }
+
+    .download-link {
```
Comment (cb341) [inline app/views/onlylogs/logs/index.html.erb:18]: info: It is a link, not a button.
I don't see why we'd want to style it as a button.

The other elements of the logs page are mostly unstyled and I'd like to do the same for the link.
The alternative would be to set a different font and color but I don't see the need to do that instead of relying on the browser defaults as we do for the other nodes.

File: test/controllers/onlylogs/logs_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module Onlylogs
+  class LogsControllerTest < ActionDispatch::IntegrationTest
+    setup do
+      Onlylogs.instance_variable_set(:@configuration, nil)
```
Comment (coorasse) [inline test/controllers/onlylogs/logs_controller_test.rb]: why is this needed? 🤔

File: test/controllers/onlylogs/logs_controller_test.rb
Comment (coorasse) [inline test/controllers/onlylogs/logs_controller_test.rb:1]: can you please add a test where you expect and error when trying to fetch a file that is a good, valid log file, in the test/fixtures/files folder, but which is not in the whitelist?

Comment (coorasse) [reply test/controllers/onlylogs/logs_controller_test.rb:1]: Ah, and also an additional test for the `.1` file. If we allow "development.log", "development.log.X" should also be allowed.

**Review by coorasse — COMMENTED** [review]
Love it! May you just add two more tests? I want to be 120% sure we cover all expected cases. Maybe take also a minute more to think if we could add another case. I don't mind having "too many tests" for this part since we should be sure. 👍 
Check also that the endpoint is fully protected when basic auth is enabled.

File: test/controllers/onlylogs/logs_controller_test.rb (hunk 3 above)
Comment (cb341) [reply test/controllers/onlylogs/logs_controller_test.rb]: Tests pass without it.
I don't think that it is used.

File: test/controllers/onlylogs/logs_controller_test.rb
Comment (cb341) [reply test/controllers/onlylogs/logs_controller_test.rb:1]: Yes, these are edge-cases I have missed.
I will add coverage for:

1. `.1`
2. basicauth
3. whitelist

Comment (cb341) [reply test/controllers/onlylogs/logs_controller_test.rb:1]: ec7d0a58b50e

File: test/fixtures/files/rotated.log.1
Comment (cb341) [inline test/fixtures/files/rotated.log.1:1]: This is a truncated version of the fixtures development log

File: app/helpers/onlylogs/application_helper.rb (hunk 1 above)
Comment (cb341) [reply app/helpers/onlylogs/application_helper.rb:3]: Ah! This allows us to hide system paths for when files are placed outside of the system dir

**Review by coorasse — APPROVED** [review]
